tests/test_pandaspysparklivyclient.py

```diff
@@ -44,15 +44,20 @@ def test_execute_sql_pandas_pyspark_livy():
     # desired pandas df
     records = [{u'buildingID': 0, u'date': u'6/1/13', u'temp_diff': 12},
                {u'buildingID': 1, u'date': u'6/1/13', u'temp_diff': 0}]
-    desired_result = pd.DataFrame(records)
+    desired_df = pd.DataFrame(records)
 
     command = "command"
     result = client.execute_sql(command)
 
     execute_m.assert_called_with('sqlContext.sql("""{}""").toJSON().take({})'.format(command, 10))
-
-    # Verify result is desired pandas df
-    assert_frame_equal(desired_result, result)
+    
+    assert isinstance(result, Result)
+    assert isinstance(result, DataFrameResult)
+    mock_shell = MagicMock()
+    df = result.render(mock_shell)
+    assert not mock_shell.write.called
```

Comment:
Assertions in 58 and 59 should ideally be done in their own unit test and not here. They should be part of a test_results class. Granularity of unit tests will allow us to quickly debug where the problem is.